### ISSUE TITLE
Initial data after streaming not work

### ISSUE THREAD

**Issue by chuankaiwei** [brillout/vite-plugin-ssr#552]
### Description

I migrate an existing project to SSR. I want to hydrate API data directly into the context to reduce migration costs. So I noticed this feature https://vite-plugin-ssr.com/stream#initial-data-after-streaming. But I got an error. Reading through the source code, I found that the `pageContext` is asserted to be an object before executing the `pageContextPromise`.

```Typescript
if (isObject(result) && !isDocumentHtml(result)) {
    assertHookResult(result, 'render', ['documentHtml', 'pageContext', 'injectFilter'] as const, renderFilePath)
  }
objectAssign(pageContext, { _renderHook: { hookFilePath: renderFilePath, hookName: 'render' as const } })

let pageContextPromise: Promise<unknown> | null = null
if (hasProp(result, 'pageContext')) {
  const pageContextProvidedByRenderHook = result.pageContext
  if (isPromise(pageContextProvidedByRenderHook)) {
    pageContextPromise = pageContextProvidedByRenderHook
  } else {
    assertPageContextProvidedByUser(pageContextProvidedByRenderHook, { hook: pageContext._renderHook })
    Object.assign(pageContext, pageContextProvidedByRenderHook)
  }
}
```

### Error Message + Error Stack

```shell
Error: [vite-plugin-ssr@0.4.57][Wrong Usage] The `pageContext` provided by the `export { render }` of /renderer/_default.page.server.js should be an object.
```
```

**Comment by brillout** [comment]
Update to the latest version. I made some changes.

**Comment by r0m4n27** [comment]
I've tried it with the newest version and it's still doesn't work. `assertHookResult` has an assertion that the `pageContent`
has to be an object.

```shell
Error: [vite-plugin-ssr@0.4.58][Wrong Usage] The `pageContext` provided by the `export { render }` of @packages/site/dist/renderer/_default.page.server.js should be an object.
    at assertPageContextProvidedByUser (/<path-to-app>/node_modules/.pnpm/vite-plugin-ssr@0.4.58_vite@4.0.1/node_modules/vite-plugin-ssr/dist/cjs/shared/assertPageContextProvidedByUser.js:13:29)
    at assertHookResult (/<path-to-app>/node_modules/.pnpm/vite-plugin-ssr@0.4.58_vite@4.0.1/node_modules/vite-plugin-ssr/dist/cjs/shared/assertHookResult.js:15:79)
    at executeRenderHook (/<path-to-app>/node_modules/.pnpm/vite-plugin-ssr@0.4.58_vite@4.0.1/node_modules/vite-plugin-ssr/dist/cjs/node/renderPage/renderPageContext.js:180:49)
    at async prerenderPageContext (/<path-to-app>/node_modules/.pnpm/vite-plugin-ssr@0.4.58_vite@4.0.1/node_modules/vite-plugin-ssr/dist/cjs/node/renderPage/renderPageContext.js:84:30)
    at async /<path-to-app>/node_modules/.pnpm/vite-plugin-ssr@0.4.58_vite@4.0.1/node_modules/vite-plugin-ssr/dist/cjs/node/prerender/runPrerender.js:325:57
```

**Comment by brillout** [comment]
Fix released in `0.4.59`.

In case your company is up for it: https://github.com/sponsors/brillout.
